commit d6cf895847f60af83bad62b15f1da14abd331fae
Author: Kent Overstreet <kent.overstreet@linux.dev>
Date:   Tue Sep 24 05:08:39 2024 -0400

    bcachefs: Remove unnecessary peek_slot()
    
    hash_lookup() used to return an errorcode, and a peek_slot() call was
    required to get the key it looked up. But we're adding fault injection
    for transaction restarts, so fix this old unconverted code.
    
    Signed-off-by: Kent Overstreet <kent.overstreet@linux.dev>

diff --git a/fs/bcachefs/fsck.c b/fs/bcachefs/fsck.c
index 6b2ddbabe3e7..c96025b8b65d 100644
--- a/fs/bcachefs/fsck.c
+++ b/fs/bcachefs/fsck.c
@@ -170,7 +170,7 @@ static int lookup_dirent_in_snapshot(struct btree_trans *trans,
 	if (ret)
 		return ret;
 
-	struct bkey_s_c_dirent d = bkey_s_c_to_dirent(bch2_btree_iter_peek_slot(&iter));
+	struct bkey_s_c_dirent d = bkey_s_c_to_dirent(k);
 	*target = le64_to_cpu(d.v->d_inum);
 	*type = d.v->d_type;
 	bch2_trans_iter_exit(trans, &iter);